EventReader/src/Read/SqlSource.cs

```diff
@@ -0,0 +1,69 @@
+﻿using System;
+using System.Collections.Generic;
+using System.Data;
+using System.Data.SqlClient;
+
+namespace EventReader.Read
+{
+    public class SqlSource : IDataSource
```

Comment:
This is pretty good.  The next step in isolating this responsibility is nesting all the SQL specific things under a separate namespace (in a larger app, this may even be a separate assembly)